src/samples/audio/smart_amp_test_ipc4.c

```diff
@@ -37,57 +55,50 @@ typedef void (*smart_amp_proc)(int8_t const *src_ptr,
 struct smart_amp_data {
 	struct sof_smart_amp_ipc4_config ipc4_cfg;
 	struct sof_smart_amp_config config;
-	struct comp_data_blob_handler *model_handler;
-	void *data_blob;
-	size_t data_blob_size;
 	smart_amp_proc process;
 	uint32_t out_channels;
 };
 
+#ifdef __MODULE_BUILD__
+static struct smart_amp_data smart_amp_priv;
+#endif
+
 static int smart_amp_init(struct processing_module *mod)
 {
 	struct smart_amp_data *sad;
-	struct comp_dev *dev = mod->dev;
 	struct module_data *mod_data = &mod->priv;
-	const size_t in_size = sizeof(struct ipc4_input_pin_format) * SMART_AMP_NUM_IN_PINS;
-	const size_t out_size = sizeof(struct ipc4_output_pin_format) * SMART_AMP_NUM_OUT_PINS;
 	int ret;
 	const struct ipc4_base_module_extended_cfg *base_cfg = mod_data->cfg.init_data;
 
-	comp_dbg(dev, "smart_amp_init()");
+	LOG_DBG("smart_amp_init()");
```

Comment:
cannot you just declare `domp_dbg()` and friends as dummies themselves for modular builds?

Reply:
No. The problem is that dev is not available for loadable module build. please check "struct processing_module".   I spent much effort to make compiling pass.

Reply:
@RanderWang sorry, don't understand. You replace `comp_dbg()` with `LOG_DBG()` and then redefine it to a NOP for loadable modules. So, why don't you just redefine the original `comp_dbg()` to a NOP? If you do, you won't care about `dev`

Reply:
@lyakh I understand your idea. if I use comp_dbg(), it will be more ifdef in the code. So I don't prefer it
````c
#ifdef  MODULE
stuct comp_dev dev = mode->dev;
#endif
.....
````c

Reply:
then define `comp_dbg()` to something, that keeps `dev` "used" and prevents compiler unused variable warnings?

Reply:
this return to the first question dev is not available for loadable module

Reply:
This should be resolved by using native Zephyr logging, but this is WIP.

Reply:
> this return to the first question dev is not available for loadable module

I mean something like
```
#define dev_dbg(dev, ...) do { (void)dev; } while (0)
```

Reply:
The previous one wouldn't work probably because `dev` isn't available at all as a variable. So I think a simple
```
#define dev_dbg(...) do {} while (0)
```
should do. If for some reason that doesn't work, maybe
```
#define dev_dbg(dev, ...) do { int dev; (void)dev; } while (0)
```
would.